src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -477,6 +477,14 @@ protected function forward(Request $request, $catch = false, Response $entry = n
             $request->headers->set('X-Forwarded-For', $request->server->get('REMOTE_ADDR'));
         }
 
+        // modify the Forwarded header if needed
+        $forwarded = $request->headers->get('Forwarded');
+        if ($forwarded) {
+            $request->headers->set('Forwarded', preg_replace('/for=([0-9a-zA-Z.,\\s]*)/', '$0, '.$request->server->get('REMOTE_ADDR'), $forwarded));
```

Comment:
the Forwarded header allows several "for" attribute, so we should just append instead of replacing, isn't it?